src/mgr/FunAction.cc

```diff
@@ -49,10 +48,12 @@ Mgr::FunAction::dump(StoreEntry* entry)
 {
     debugs(16, 5, MYNAME);
     Must(entry != nullptr);
-    if (UsingSmp())
-        storeAppendPrintf(entry, "by kid%d {\n", KidIdentifier);
+
+    // always non-aggregatable, no need to check
+    openKidSection(entry, is_yaml());
+
     handler(entry);
-    if (atomic() && UsingSmp())
-        storeAppendPrintf(entry, "} by kid%d\n\n", KidIdentifier);
-}
 
+    // always non-aggratable and atomic, no need to check
+    closeKidSection(entry, is_yaml());
```

Comment:
N.B. This PR code is subject to the same "always non-aggratable" concerns documented in the other change request. The suggestion below applies the second suggestion from that change request to this code as well.

FunActions are not always atomic(). For example, mgr:objects action (i.e. stat_objects_get) is not atomic. For more information, see Mgr::Action::atomic() implementation inherited by FunAction.

```suggestion
    if (!aggregatable() && atomic())
        closeKidSection(entry, is_yaml());
    // else: non-aggregatable() non-atomic() actions call closeKidSection() when they are done
```

P.S. The comment suggested above does document a dependency, similar to the comment rejected in another change request. However, in this case, the comment is warranted because this dependency cannot be "trivially made explicit in code" (unfortunately).

Reply:
added the comment, slightly reworded